src/ast_editor.ml

```diff
@@ -321,9 +327,19 @@ let open_pp_doc instance ~document =
   match fetch_pp_code ~document with
   | Error e -> Promise.return (Error e)
   | Ok pp_pp_str ->
+    let file_name =
+      match Pp_path.get_kind ~document with
+      | Structure `Reason ->
+        String.chop_suffix_exn ~suffix:".re" (TextDocument.fileName document)
+        ^ ".ml"
+      | Signature `Reason ->
+        String.chop_suffix_exn ~suffix:".rei" (TextDocument.fileName document)
+        ^ ".mli"
+      | _ -> TextDocument.fileName document
+    in
```

Comment:
In nthe future we can use the refmt executable to convert it from ocaml to reasonml.